core/lib/spree/event/configuration.rb

```diff
@@ -14,12 +18,64 @@ def autoload_subscribers
       end
 
       def adapter
-        @adapter ||= Spree::Event::Adapters::ActiveSupportNotifications
+        @adapter ||= Spree::Core::VersionedValue.new(
+          Spree::Event::Adapters::ActiveSupportNotifications,
+          "4.0.0.alpha" => Spree::Event::Adapters::EventBus.new
+        ).call.tap do |value|
+          deprecate_if_legacy_adapter(value)
+        end
       end
 
+      # Only used by {Spree::Event::Adapters::ActiveSupportNotifications}.
       def suffix
         @suffix ||= '.spree'
```

Comment:
Do you think we should display an error/refuse booting if the suffix is set even though the new adapter is being used?

Reply:
I think it'd be too aggressive as the value is set by default. We could render a warning, but still, I'm not convinced. We'd be forcing users to add `config.events.suffix = nil` to their initializer because of a deprecated behavior. On top of that, we'd bother them again to remove it once the legacy adapter is removed.